### PR TITLE
Add cluster-autoscaler docs

### PR DESCRIPTION
**Pull request by jonathanmeier5**
*Issue #, if available:*
Epic: https://github.com/aws/eks-anywhere/issues/3186
Ticket: https://github.com/aws/eks-anywhere/issues/3717

*Description of changes:*
Adds documentation for cluster-autoscaler package

*Testing (if applicable):*
Ran `make serve` locally. Clicked through links and verified pages existed where they should.

Happy to do any other recommended manual testing.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+## Uninstall
+
+To uninstall Cluster Autoscaler, simply delete the package
+
+```bash
+eksctl anywhere delete package --cluster <cluster-name> cluster-autoscaler
+```
+
+## Installing Cluster Autoscaler on Workload Cluster
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md:73]:
```suggestion
## Installing Cluster Autoscaler on workload cluster
```

**Review by chrisnegus — COMMENTED** [review]
Only one small suggestion.

File: docs/content/en/docs/reference/packagespec/cluster-autoscaler/v9.21.0.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+title: "v9.21.0"
+linkTitle: "v9.21.0"
+weight: 20
+description: >
+---
+
+### Configuring Cluster Autoscaler in EKS Anywhere package spec
+
+| Parameter | Description | Default |
```
Comment (ivyostosh) [inline docs/content/en/docs/reference/packagespec/cluster-autoscaler/v9.21.0.md:10]: Just to confirm - all parameters here have been tested by us?

File: docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+{{< content "../prereq.md" >}}
+
+
+## Install
+
+<!-- this content needs to be indented so the numbers are automatically incremented -->
+1. Generate the package configuration
+   ```bash
+   eksctl anywhere generate package cluster-autoscaler --source cluster > cluster-autoscaler.yaml
```
Comment (ivyostosh) [inline docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md:18]: I think this is outdated, but we will do a bulk update, so feel free to leave it. Commenting just as FYI.

**Review by ivyostosh — COMMENTED** [review]
Did you run a local build specified [here](https://github.com/aws/eks-anywhere/tree/main/docs) and confirm all links work?

File: docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md (hunk 3 above)
Comment (TerryHowe) [reply docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md:18]:
```suggestion
   eksctl anywhere generate package cluster-autoscaler --cluster clusterName > cluster-autoscaler.yaml
```

File: docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+First, retrieve the management cluster's kubeconfig secret:
+```yaml
+kubectl -n eksa-system get secrets <management-cluster-name>-kubeconfig -o yaml > mgmt-secret.yaml
+```
+
+Update the secret's namespace to the namespace in the workload cluster that you would like to deploy the cluster autoscaler to.
+Then, apply the secret to the workload cluster.
+```yaml
+kubectl --kubeconfig /path/to/workload/kubeconfig apply -f mgmt-secret.yaml
```
Comment (TerryHowe) [inline docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md:85]: Bummer the workload clusters needs to auth to the management cluster. There is no way to least privilege this to something other than full admin?

File: docs/content/en/docs/reference/packagespec/cluster-autoscaler/v9.21.0.md (hunk 2 above)
Comment (jonathanmeier5) [reply docs/content/en/docs/reference/packagespec/cluster-autoscaler/v9.21.0.md:10]: I double checked everything and noticed that `imagePullSecrets` seems to be overwritten by the controller. I've pulled it out of the docs. I think I also have a mismatch between the schema.json and values.yaml for the value. I'll open a PR to resolve it as well.

File: docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md (hunk 4 above)
Comment (jonathanmeier5) [reply docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md:85]: Unfortunately this is our only option right now. This was before I joined, but it is discussed in the [design doc](https://github.com/aws/eks-anywhere/blob/195ef39623a32170fd7837813a03c03f2df14132/designs/autoscaler.md#L274-L274).

**Comment by jonathanmeier5** [comment]
> Did you run a local build specified [here](https://github.com/aws/eks-anywhere/tree/main/docs) and confirm all links work?

Yes, I've clicked through the links and confirmed they work using both `make serve` and `make container-serve`.

**Comment by ivyostosh** [comment]
Sorry one more thing, could you add an update command similar to this [PR](https://github.com/aws/eks-anywhere/commit/938d1a2a9f68817c0f2ec462f0dc2098106856e1) as well? I tried to do a bulk update for all packages yesterday, but cluster-autoscaler is not there yet.

**Comment by jonathanmeier5** [comment]
> Sorry one more thing, could you add an update command similar to this [PR](https://github.com/aws/eks-anywhere/commit/938d1a2a9f68817c0f2ec462f0dc2098106856e1) as well? I tried to do a bulk update for all packages yesterday, but cluster-autoscaler is not there yet.

Done. Verified docs display correctly after the change.

File: docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+   ```
+
+   Example command output
+   ```
+   NAMESPACE                  NAME                          PACKAGE              AGE   STATE       CURRENTVERSION                                               TARGETVERSION                                                         DETAIL
+   eksa-packages-mgmt-v-vmc   cluster-autoscaler            cluster-autoscaler   18h   installed   9.21.0-1.21-147e2a701f6ab625452fe311d5c94a167270f365         9.21.0-1.21-147e2a701f6ab625452fe311d5c94a167270f365 (latest)
+   ```
+
+## Update
+To update package configuration, update adot.yaml file, and run the following command:
```
Comment (ivyostosh) [inline docs/content/en/docs/tasks/packages/cluster-autoscaler/_index.md:61]:
```suggestion
To update package configuration, update cluster-autoscaler.yaml file, and run the following command:
```

**Pull request merged**